apps/src/weblab/brambleHost.js

```diff
@@ -392,49 +392,62 @@ function onInspectorChanged(callback) {
   onInspectorChangedCallback_ = callback;
 }
 
+var _hasMounted = false;
+
+function ensureMounted() {
+  if (!_hasMounted) {
+    bramble_.mount(`${weblabRoot}/${currentProjectPath}`);
+  }
+  _hasMounted = true;
+}
+
 function startInitialFileSync(callback, forceResetToStartSources) {
-  const currentProjectVersion = webLab_.getCurrentFilesVersionId();
 
-  if (!currentProjectVersion || forceResetToStartSources) {
-    // Get initial sources to put in file system
-    const startSources = webLab_.getStartSources();
+  ensureProjectRootDirExists(err => {
+    const wrappedCallback = err => {
+      ensureMounted();
+      callback(err);
+    };
 
-    // Sync what's on the service locally (should be nothing, but this ensures
-    // our _lastSyncedVersionId matches before we make changes)
-    syncFiles(err => {
-      if (err) {
-        console.warn(`Bramble host: Initial syncFiles failed with error: ${err}`);
-      }
-      // put the source files into the Bramble file system
-      putFilesInBramble(startSources, err => {
-        if (err) {
-          callback();
-        } else {
-          if (forceResetToStartSources) {
-            console.log('startInitialFileSync: forceResetToStartSources - calling syncFiles()');
-            // Start the initial files sync
-            syncFiles(callback);
-          } else {
-            // First-time level load - no project versionId
+    if (err) {
+      wrappedCallback(err);
+      return;
+    }
 
-            // Ignore all of the change events that just occured as we wrote the
-            // startSources into bramble - there is no need to save these during
-            // sync until a user-initiated change occurs
-            resetBrambleChangesAndProjectVersion(currentProjectVersion);
-            webLab_.registerBeforeFirstWriteHook(uploadAllFilesFromBramble);
+    const currentProjectVersion = webLab_.getCurrentFilesVersionId();
+    if (!currentProjectVersion || forceResetToStartSources) {
+      // Get initial sources to put in file system
+      const startSources = webLab_.getStartSources();
 
-            // Start the initial files sync
-            syncFiles(callback);
-          }
+      // Sync what's on the service locally (should be nothing, but this ensures
+      // our _lastSyncedVersionId matches before we make changes)
+      syncFiles(err => {
+        if (err) {
+          console.warn(`Bramble host: Initial syncFiles failed with error: ${err}`);
         }
+        // put the source files into the Bramble file system
+        putFilesInBramble(startSources, err => {
+          if (!err) {
+            if (forceResetToStartSources) {
+              console.log('startInitialFileSync: forceResetToStartSources - calling syncFiles()');
```

Comment:
This has become an awkward structure: This `if` case only contains the `console.log` call.  I think the statement being logged still accurately represents the behavior of that case, but I can't help feeling like these conditionals should collapse now, making it obvious that we have only two paths through this logic.

```js
putFilesInBramble(startSources, (err) => {
  if (!err && !forceResetToStartSources) {
    resetBrambleChangesAndProjectVersion(currentProjectVersion);
    webLab_.registerBeforeFirstWriteHook(uploadAllFilesFromBramble);
  }
  syncFiles(wrappedCallback);
});
```

Bonus points if you can come up with a helpful name for the `!err && !forceResetToStartSources` condition. :grin: